src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParametersUpdatedHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+
+namespace Microsoft.Health.Fhir.Core.Features.Definition
+{
+    public class SearchParametersUpdatedHandler : INotificationHandler<SearchParametersHashUpdated>, INotificationHandler<ReindexJobCompleted>
+    {
+        private SearchParameterDefinitionManager _searchParameterDefinitionManager;
```

Comment:
can this be readonly? And any reason we don't want to use the interface here?

Reply:
Yes, it can be readonly, I will update that.  I am using the concrete class rather than the interface because the interface doesn't expose the ability to set the SearchParametersHash.  I didn't want to add that to the interface, because that should only be able to be updated here, by the handler.  All other classes that access the SearchParameterDefinition manager using the interface won't have the set access.